inner-scripts/kube-bind

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+space_name=""
+resource_to_bind=""
+start_konnector=false
+cm_namespace=kubestellar # the namespace where the configmaps live in on the provider side
+
+while (( $# > 0 )); do
+    case "$1" in
+    (-h|--help)
+        echo "Usage: $0 (--start_konnector boolean | -X)* space_name resource_to_bind"
+        echo "space_name is the name of a kube-bind consumer space, which currently must be a kcp workspace whose parent is root"
+        echo "resource_to_bind is a resource offered by the kube-bind provider, in lower case and plural form"
+        exit 0;;
+    (-X)
+        set -o xtrace;;
+    (--start-konnector)
+        if (( $# >1 ))
+        then start_konnector="$2"; shift
+        else echo "$0: missing start-konnector value" >&2; exit 1
+        fi;;
+    (-*)
+        echo "$0: flag syntax error" >&2
+        exit 1;;
+    (*)
+        if [ -z "$space_name" ]
+        then space_name="$1"
+        elif [ -z "$resource_to_bind" ]
+        then resource_to_bind="$1"
+        else echo "$0: too many positional arguments" >&2
+        exit 1
+        fi
+    esac
+    shift
+done
+
+if [ -z "$space_name" ]
+then
+    echo "space_name not specified" >&2
+    exit 1
+fi
+
+if [ -z "$resource_to_bind" ]
+then
+    echo "resource_to_bind not specified" >&2
+    exit 1
+fi
+
+case "$start_konnector" in
+    (true|false) ;;
+    (*) echo "$0: start_konnector should be true or false" >&2
+	  exit 1;;
+esac
+
+initial_ws=$(kubectl ws . --short)
+
```

Comment:
```
trap "kubectl ws use $initial_ws &> /dev/null # go back to the initial state" EXIT
```

instead of doing the ws change at the end of the script. The trap makes this happen even if the script exits early due to an error.

Reply:
Thanks for the guidance, will improve in the next PR.